### PR TITLE
Dps Fetch Refactor #007: Extend utils with new functionality

### PR DESCRIPTION
**Pull request by haakonvt**
### 007 🥂

### CODE REVIEW

**Review by erlendvollset — COMMENTED** [review]
I wrote "good candidate for some docstring tests" on a few methods there, but they can be ofc just be regular unit tests aswell if you prefer that.

File: cognite/client/utils/_time.py [hunk 1]
```diff
@@ -142,3 +135,31 @@ def convert_time_attributes_to_datetime(item: Union[Dict, List[Dict]]) -> Union[
...
+        start -= remainder
+    gms = granularity_to_ms(granularity)
+    remainder = (end - start) % gms
+    if remainder:
+        # Ceil `end` when not exactly at boundary decided by `start + N * granularity`
+        end += gms - remainder
+    return start, end
+
+
+def split_time_range(start: int, end: int, n_splits: int, granularity_in_ms: int) -> List[int]:
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/utils/_time.py:154]: See #1047

File: cognite/client/utils/_time.py [hunk 2]
```diff
@@ -142,3 +135,31 @@ def convert_time_attributes_to_datetime(item: Union[Dict, List[Dict]]) -> Union[
             new_items.append(_convert_time_attributes_in_dict(el))
         return new_items
     raise TypeError("item must be dict or list of dicts")
+
+
+def align_start_and_end_for_granularity(start: int, end: int, granularity: str) -> Tuple[int, int]:
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/utils/_time.py:140]: See #1047

File: cognite/client/utils/_auxiliary.py [hunk 3]
```diff
@@ -146,6 +150,11 @@ def __bool__(self) -> bool:
         return len(self.__heap) > 0
 
 
+def split_into_n_parts(seq: Sequence[T], /, n: int) -> Iterator[Sequence[T]]:
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/utils/_auxiliary.py:153]: On it ⚡

File: cognite/client/utils/_auxiliary.py [hunk 4]
```diff
@@ -173,3 +182,8 @@ def convert_true_match(true_match: Union[dict, list, Tuple[Union[int, str], Unio
         return true_match
     else:
         raise ValueError("true_matches should be a dictionary or a two-element list: found {}".format(true_match))
+
+
+def find_duplicates(seq: Sequence[T]) -> Set[T]:
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/utils/_auxiliary.py:187]: On it 💡

File: cognite/client/utils/_auxiliary.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/utils/_auxiliary.py:153]: Done! https://github.com/cognitedata/cognite-sdk-python/pull/1047/files#diff-1d2462c0f2a2dfee857bfde4966c1f353c6555def8165897cf15ff064fdea090

File: cognite/client/utils/_auxiliary.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/utils/_auxiliary.py:187]: Done! https://github.com/cognitedata/cognite-sdk-python/pull/1047/files#diff-1d2462c0f2a2dfee857bfde4966c1f353c6555def8165897cf15ff064fdea090

**Review by erlendvollset — APPROVED** [review]

**Pull request merged**